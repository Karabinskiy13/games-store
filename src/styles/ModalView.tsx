import styled from 'styled-components';
import { Box } from '@mui/material';

export const StyledModal = styled(Box)`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  max-width: 1200px;
  max-height: 1000px;
  bgcolor: black;
  border: 2px solid black;
  box-shadow: 24;
  p: 4;
`;

export const ModalImage = styled.img`
  width: 100%;
  max-height: 800px;
  @media (max-width: 500px) {
    width: 100vw;
  }
`;
