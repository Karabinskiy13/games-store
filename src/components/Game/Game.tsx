import React from 'react';
import Link from 'next/link';
import Image from 'next/image';

import { Games } from '../../../types';
import { Button, GameItem } from '../../styles';

const Game = ({ game, showModal }: Games) => {
  const posterLink = `http://${game.fields.poster?.fields.file.url}`;
  const showDescription = `/games/${game.fields.name}`;

  return (
    <GameItem>
      <Image
        className="poster"
        src={posterLink}
        width={270}
        height={400}
        alt="poster"
        data-testId="poster"
        onClick={() => showModal(true)}
      />
      <div className="name">{game.fields.name}</div>
      <div data-testId="price">{game.fields.price}UAH</div>
      <div data-testId="releaseDate">Release Date: {game.fields.releaseDate}</div>
      <Link href={showDescription}>
        <Button data-testId="button" className="button">
          {game.fields.showFull}
        </Button>
      </Link>
    </GameItem>
  );
};

export default Game;
