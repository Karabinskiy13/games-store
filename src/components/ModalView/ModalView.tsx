import React from 'react';

import { Modal } from '@mui/material';
import { ModalImage, StyledModal } from '../../styles';

export type ModalProps = {
  show: boolean;
  name?: string;
  url?: string;
  hideModal: () => void;
};

const ModalView = ({ show, url, name, hideModal }: ModalProps) => {
  return (
    <Modal
      open={show}
      data-Testid="modalWrapper"
      onClose={() => hideModal()}
      aria-describedby="modal-modal-picture">
      <StyledModal>
        <ModalImage className="modal__image" alt="modal__image" src={url} />
        <div>{name}</div>
      </StyledModal>
    </Modal>
  );
};

export default ModalView;
