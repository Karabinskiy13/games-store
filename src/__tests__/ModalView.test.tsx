import React from 'react';
import { jest } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Game from '../components/Game/Game';
import ModalView, { ModalProps } from '../components/ModalView/ModalView';

const modalProps: ModalProps = {
  show: true,
  name: 'name',
  url: '/image.png',
  hideModal: jest.fn(() => false)
};

describe('<ModalView>', () => {
  test('Should render component', () => {
    const { asFragment } = render(<ModalView {...modalProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should display modal view from props', async () => {
    render(<ModalView {...modalProps} />);
    await screen.findAllByAltText('modal__image');
    expect(screen.getByAltText('modal__image')).toHaveAttribute('src', '/image.png');
  });
});
