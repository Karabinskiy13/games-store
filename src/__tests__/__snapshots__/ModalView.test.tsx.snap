// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SinglePicture> Should render component 1`] = `<DocumentFragment />`;
